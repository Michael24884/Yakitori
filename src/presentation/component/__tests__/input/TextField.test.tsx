/**
 * @format
 */
import React from 'react';
import {Text, Image, TextInputProps} from 'react-native';

import {TextField} from '@components';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('Test TextView', () => {
  it('render correctly', () => {
    const instance = renderer.create(<TextField />);
    expect(instance.toJSON()).toMatchSnapshot();
  });
  it('Test render prefix', () => {
    const instance = renderer.create(<TextField />);
    expect(instance.root.findAllByType(Image)).toHaveLength(0);

    // test render Prefix
    instance.update(<TextField prefix={<Text children="test" />} />);
    expect(instance.root.findByProps({children: 'test'})).toBeDefined();
    expect(instance.root.findAllByType(Image)).toHaveLength(0);

    // test render Prefix
    const prefixIcon = 1;
    instance.update(<TextField prefixIcon={prefixIcon} />);
    expect(instance.root.findByType(Image)?.props?.source).toEqual(prefixIcon);
  });

  it('Test render suffix', () => {
    const instance = renderer.create(<TextField />);
    expect(instance.root.findAllByType(Image)).toHaveLength(0);

    // test render Prefix
    instance.update(<TextField suffix={<Text children="test" />} />);
    expect(instance.root.findByProps({children: 'test'})).toBeDefined();
    expect(instance.root.findAllByType(Image)).toHaveLength(0);

    // test render Prefix
    const prefixIcon = 1;
    instance.update(<TextField suffixIcon={prefixIcon} />);
    expect(instance.root.findByType(Image)?.props?.source).toEqual(prefixIcon);
  });

  it('Test render TextInput with props', () => {
    const inputProps: TextInputProps = {value: 'test'};
    const instance = renderer.create(<TextField inputProps={inputProps} />);
    expect(instance.root.findByProps(inputProps)).toBeDefined();
  });
});
