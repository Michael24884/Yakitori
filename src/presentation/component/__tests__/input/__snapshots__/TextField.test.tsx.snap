// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TextView render correctly 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "width": 16,
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <Text
    style={
      Object {
        "color": "#4f4f4f",
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#F1F3F8",
        "height": 1,
        "marginTop": 8,
        "width": "100%",
      }
    }
  />
</View>
`;
